File: stasis-core/src/de/boxxit/stasis/serializer/MessageExceptionSerializer.java
package de.boxxit.stasis.serializer;

import com.esotericsoftware.kryo.Kryo;
import com.esotericsoftware.kryo.Serializer;
import com.esotericsoftware.kryo.io.Input;
import com.esotericsoftware.kryo.io.Output;
import de.boxxit.stasis.MessageException;

/**
 * User: Christian Fruth
 */
public class MessageExceptionSerializer extends Serializer<MessageException>
{
	public MessageExceptionSerializer()
	{
	}

	@Override
	public void write(Kryo kryo, Output output, MessageException object)
	{
		kryo.writeObjectOrNull(output, object.getId(), String.class);
		kryo.writeObjectOrNull(output, object.getMessage(), String.class);
	}

	@Override
	public MessageException read(Kryo kryo, Input input, Class<MessageException> type)
	{
		String id = kryo.readObjectOrNull(input, String.class);
		String message = kryo.readObjectOrNull(input, String.class);
		return new MessageException(id, message);
	}
}

File: stasis-ios/Stasis/MessageExceptionSerializer.m
//
//  SerializableExceptionSerializer.m
//  Stasis
//
//  Created by Christian Fruth on 06.03.13.
//  Copyright (c) 2013 Boxx IT Solutions e.K. All rights reserved.
//

#import "MessageExceptionSerializer.h"
#import "MessageException.h"
#import "Kryo.h"

@implementation MessageExceptionSerializer

- (BOOL)acceptsNull
{
	return NO;
}

- (void)write:(Kryo *)kryo value:(id)value to:(KryoOutput *)output
{
	MessageException *exception = value;
	[kryo writeNullableObject:exception.name withClass:[NSString class] to:output];
	[kryo writeNullableObject:exception.reason withClass:[NSString class] to:output];
}

- (id)read:(Kryo *)kryo withClass:(Class)type from:(KryoInput *)input
{
	NSString *ident = [kryo readNullableObject:input ofClass:[NSString class]];
	NSString *message = [kryo readNullableObject:input ofClass:[NSString class]];
	return [MessageException exceptionWithIdent:ident andMessage:message];
}

@end
